shared-bindings/displayio/Display.c

```diff
@@ -356,17 +373,132 @@ const mp_obj_property_t displayio_display_bus_obj = {
 };
 
 
+
+
+#include "py/objarray.h"
+mp_obj_array_t *array_new(char typecode, size_t n);
+mp_obj_t array_subscr(mp_obj_t self_in, mp_obj_t index_in, mp_obj_t value);
+
+//|   .. method:: fill_area(x, y, w, h)
+//|
+//|     Switches to displaying the given group of layers. When group is None, the default
+//|     CircuitPython terminal will be shown.
+//|
+//|     :param int x: The left edge of the area
+//|     :param int y: The top edge of the area
+//|     :param int w: The width of the area
+//|     :param int h: The height of the area
+STATIC mp_obj_t displayio_display_obj_fill_area(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_args) {
+  enum { ARG_x, ARG_y, ARG_width, ARG_height, ARG_buffer };
+    static const mp_arg_t allowed_args[] = {
+      { MP_QSTR_x, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = -1} },
+        { MP_QSTR_y, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = -1} },
+        { MP_QSTR_width, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = -1} },
+        { MP_QSTR_height, MP_ARG_INT | MP_ARG_KW_ONLY, {.u_int = -1} },
+        { MP_QSTR_buffer, MP_ARG_OBJ | MP_ARG_KW_ONLY, {} },
+    };
+    mp_arg_val_t args[MP_ARRAY_SIZE(allowed_args)];
+    mp_arg_parse_all(n_args - 1, pos_args + 1, kw_args, MP_ARRAY_SIZE(allowed_args), allowed_args, args);
+
+    displayio_display_obj_t *self = native_display(pos_args[0]);
+    mp_int_t x = args[ARG_x].u_int;
+    mp_int_t y = args[ARG_y].u_int;
+    mp_int_t w = args[ARG_width].u_int;
+    mp_int_t h = args[ARG_height].u_int;
+    mp_obj_array_t *result = (mp_obj_array_t *)(args[ARG_buffer].u_obj);
```

Comment:
I would recommend using ``mp_get_buffer_raise`` to get the buffer and its size, then you don't need all those functions for accessing the array elements, etc.

Reply:
You can see an example at https://github.com/adafruit/circuitpython/pull/2032/files#diff-28511b511b1f39aef03beef1a21a603eR82-R87